### PR TITLE
add buttons components

### PR DESCRIPTION
**Pull request by piensaenpixel**
@javierarce could you take a look
image
image

### CODE REVIEW

**Comment by javierarce** [comment]
Markup looks great, but I have a doubt regarding the colors: we are saying that `.CDB-Button--primary` is going to have a blue background, but maybe that won't be the case in certain contexts (in an alert message the primary button could be red, to inform the user that a destructive action is going to take place). Wouldn't it be better to define the colors using a theme (or a different className)?

**Comment by piensaenpixel** [comment]
But in this case, should we add a modifier? is-alert or similar?

File: src/scss/cdb-components/buttons.scss [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+@import '../cdb-variables/sizes';
+@import '../cdb-variables/colors';
+
+
+.CDB-Button {
+  padding: 8px 20px;
+  border-radius: 4px;
+  cursor: pointer;
+
+  &.is-mini {
```
Comment (xavijam) [inline src/scss/cdb-components/buttons.scss]: IMHO, it should be a modifier, not a "state".

Comment (xavijam) [inline src/scss/cdb-components/buttons.scss]:
``` css
.CDB-Button--small { }
```

?

File: src/scss/cdb-components/buttons.scss [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+  <button class="CDB-Button CDB-Button--primary is-mini is-disabled">
+    <span class="CDB-Button-Text Text is-semibold Size-small">SAVE</span>
+  </button>
+</div>
+
+```
+*/
+
+@import '../cdb-utilities/mixins';
```
Comment (xavijam) [inline src/scss/cdb-components/buttons.scss:52]: If we don't need `mixins` or `sizes`, we should not import them.

File: src/scss/cdb-components/buttons.scss (hunk 1 above)
Comment (piensaenpixel) [inline src/scss/cdb-components/buttons.scss]: +1 fixed

File: src/scss/cdb-components/buttons.scss (hunk 2 above)
Comment (piensaenpixel) [inline src/scss/cdb-components/buttons.scss:52]: mixins we'll use for transitions in hover, and sizes i've just pushed a change using the vars

Comment (xavijam) [inline src/scss/cdb-components/buttons.scss:52]: If we are not using any mixin we shouldn't import it. If we need it in the future let's face it at that time.

File: src/scss/cdb-components/buttons.scss [hunk 3]
```diff
@@ -0,0 +1,165 @@
+// Buttons styles
+// ----------------------------------------------
+
+/* SG
+# Buttons/Primary
+
```
Comment (xavijam) [inline src/scss/cdb-components/buttons.scss:6]: Could we add a description explaining which state classes we use (for example) or other relevant information related with these new buttons?

**Comment by xavijam** [comment]
IMHO I like what @piensaenpixel is proposing, it is following the same "concept" we are using for the current editor:

screen shot 2016-01-26 at 13 01 20

And seems like it is working properly. Unfortunately we can't create themes for shared components, from my point of view.

**Comment by javierarce** [comment]
Ok, then :) Let's go with the implemented solution, :+1:

File: src/scss/cdb-components/buttons.scss (hunk 3 above)
Comment (piensaenpixel) [inline src/scss/cdb-components/buttons.scss:6]: fixed @xavijam

**Pull request merged**